### PR TITLE
Add SPI documentation.

### PR DESCRIPTION
**Pull request by ntoll**
The docs as previously seen with corrections.

I'll add an issue about checking *all* the docs for * kwargs - I think it probably easier to do it all in a single PR.

### CODE REVIEW

File: docs/spi.rst [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    | SPI Mode | Polarity | Phase |
+    +==========+==========+=======+
+    | 0        | 0        | 1     |
+    +----------+----------+-------+
+    | 1        | 0        | 0     |
+    +----------+----------+-------+
+    | 2        | 1        | 1     |
+    +----------+----------+-------+
+    | 3        | 1        | 0     |
+    +----------+----------+-------+
```
Comment (dpgeorge) [inline docs/spi.rst]: The phase is actually opposite to what you have: mode=0 is CPOL=0, CPHA=0, etc.

Comment (ntoll) [inline docs/spi.rst]: OK... this was taken from the Wikipedia article you referenced. Just to be certain, can you let me know precisely what the values should be. Happy to do the leg-work.

Comment (ntoll) [inline docs/spi.rst]: Also, is it more correct to say ``CPOL`` and ``CPHA``..? I used "normal" English so numpties like me don't get blinded by TLAs. Having said that I think it important we perhaps use both in this case.

Comment (dpgeorge) [inline docs/spi.rst]: It should be:
```
0 0 0
1 0 1
2 1 0
3 1 1
```
Ideally we'd actually describe what the numbers mean:  Polarity (aka CPOL) 0 means that the clock sits at logic level 0 when idle, and goes high when active; polarity 1 means the clock sits at logic level 1 when idle.  Phase (aka CPHA) 0 means that data is sampled on the leading edge of the clock, and 1 means on the trailing edge.

Comment (ntoll) [inline docs/spi.rst]: OK... update incoming in within minutes.

**Comment by ntoll** [comment]
Updated and squashed. Is it OK?

**Comment by ntoll** [comment]
**HOLD OFF** RST is munging the table. :-(

**Comment by ntoll** [comment]
OK... good to go... sorry. I made a minor change just after testing the rendering. That'll teach me.

**Comment by dpgeorge** [comment]
Looks good!

**Comment by dpgeorge** [comment]
Yay!

**Comment by ntoll** [comment]
Yay!

**Comment by ntoll** [comment]
What's left for 1.0..? I suspect just #182..?

**Pull request merged**